canopy_zpd_mod.F90

```diff
@@ -0,0 +1,84 @@
+module canopy_zpd_mod
+
+    implicit none
+
+contains
+
+!:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+    SUBROUTINE CANOPY_ZPD( ZTOTHC, FAFRACK, FAFRACK0, UBZREF, Z0GHCM, &
+        LAMDARS, CDRAG, PAI, d_h, zo_h )
+
+!-----------------------------------------------------------------------
+
+! Description:
+!     computes zero plane displacement height and total surface roughness length basd on Massman et al. (2017).
+
+! Preconditions:
+!     foliage characteristics, drag, and in-canopy plant distribution functions
+
+! Subroutines and Functions Called:
+
+! Revision History:
+!     Prototype 06/22 by PCC, based on Massman et al. (2017) algorithms
+!     Jun 2022 P.C. Campbell: Initial standalone canopy wind model
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+        use canopy_const_mod, ONLY: rk, vonk    !constants for canopy models
+        use canopy_utils_mod     !utilities for canopy models
+
+! Arguments:
+!     IN/OUT
+        REAL(RK),    INTENT( IN )  :: ZTOTHC  ( : )   ! SUB-CANOPY Total z/h layers (nondimensional)
+        REAL(RK),    INTENT( IN )  :: FAFRACK ( : )   ! SUB-CANOPY Fractional (z) shapes of the
+        ! plant surface distribution at z=0 (nondimensional)
+        REAL(RK),    INTENT( IN )  :: FAFRACK0        ! Fractional (z) shapes of the
```

Comment:
Agreed.  There is a typo in the first argument for FAFRACK (:), where I have removed the "at z=0 (nondimensional)".  Agreed also that FAFRACK0 is technically zero since there is no cumulative LAI here.  This corresponds to Eq. 12 and 15 in Massman et al., 2017.   

Also agree to remove FAFRACK0 since its not needed as the entire FAFRACK array is being passed in.  Also changed FAFRACK to FCLAI in this module per previous comments.  Changes are found in next [commit](https://github.com/noaa-oar-arl/canopy-app/pull/20/commits/048759144984452b20ada04c065064e37bad149f) if you agree to resolve conversation.

Reply:
👍